File: src/main/java/com/safjnest/Commands/Misc/PingCommand.java
package com.safjnest.Commands.Misc;

import java.util.Collections;

import com.jagrosh.jdautilities.command.SlashCommand;
import com.jagrosh.jdautilities.command.SlashCommandEvent;

import net.dv8tion.jda.api.interactions.commands.OptionType;
import net.dv8tion.jda.api.interactions.commands.build.OptionData;

public class PingCommand extends SlashCommand {
    
    public PingCommand() {
        this.name = "ping";
        this.options = Collections.singletonList(new OptionData(OptionType.USER, "text", "The text to say.").setRequired(true));
        this.help = "Performs a ping to see the bot's delay";
    }
    
    @Override
    public void execute(SlashCommandEvent event) {
        
        long time = System.currentTimeMillis();
        event.deferReply().queue(
            hook -> hook.editOriginalFormat("Pong: %d ms " + event.getOption("text").getAsUser().getName(), System.currentTimeMillis() - time).queue()
        );
    }
}

File: src/main/kotlin/com/safjnest/commands/misc/PingCommand.kt
package com.safjnest.commands.misc

import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent
import net.dv8tion.jda.api.hooks.ListenerAdapter

/**
 * Simple Ping command converted to Kotlin
 * Shows the bot's latency
 * 
 * @author <a href="https://github.com/Leon412">Leon412</a>
 * @author Converted to Kotlin
 */
class PingCommand : ListenerAdapter() {
    
    override fun onSlashCommandInteraction(event: SlashCommandInteractionEvent) {
        if (event.name == "ping") {
            val time = System.currentTimeMillis()
            
            event.deferReply().queue { hook ->
                val ping = System.currentTimeMillis() - time
                val gatewayPing = event.jda.gatewayPing
                
                hook.editOriginal(
                    "🏓 Pong!\n" +
                    "**Response Time:** ${ping}ms\n" +
                    "**Gateway Ping:** ${gatewayPing}ms"
                ).queue()
            }
        }
    }
}